instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py

```diff
@@ -289,6 +336,8 @@ def __call__(self, environ, start_response):
             context_getter=wsgi_getter,
             attributes=collect_request_attributes(environ),
         )
+        if span.kind == trace.SpanKind.SERVER:
```

Comment:
Why is this needed? I thought feature is applicable to server instrumentations irrespective of span kind.

Reply:
In case of multiple server instrumentation (an app instrumented with both wsgi and flask/django) this check will avoid collection of custom headers in multiple spans.

Reply:
What should be the behaviour if the upstream server instrumentation injects some new headers which should be captured in downstream server instrumentation?

Reply:
Thats a good point. I didn't know if we allow user to add new headers during instrumentation.
If that is allowed, I think in that case we should let both spans to collect headers to avoid loss of info. 
There was a brief discussion with on slack for adding SERVER span check a while back [link](https://cloud-native.slack.com/archives/C01PD4HUVBL/p1636545699054600?thread_ts=1630559003.013900&cid=C01PD4HUVBL)
Please share your thoughts @srikanthccv @owais for the same.

Reply:
I am just thinking out loud here. We can revisit this if this becomes a problem for user.